### PR TITLE
Whitepaper contract

### PR DESCRIPTION
**Pull request by velzevur**
This is the contract that is intended to go in the whitepaper as a showcase for using a State Channel's off-chain Smart Contract using on-chain data. The dataitself is represented by an on-chain weather Oracle that publishes on-chain if there had been an hailstorm at a certain block height. This is later on used in the off-chain contract to trigger an insurance coverage event.

The Oracle's data publishing lifecycle requires a `query` to precede the `response`. This is either to be done in two steps as distinct transactions or as a single contract call as with the second smart contract provided. Note that there is no tests coverage for it.

### CODE REVIEW

File: test/contracts/channel_whitepaper_example.aes [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      put(state{ active_to = state.active_to + period })
+
+  stateful entrypoint withdraw(amount: int) =
+    if (Call.caller != Contract.creator)
+      abort("not_service_provider")
+    if (Chain.block_height =< state.active_to) // insured
+      if (Contract.balance - amount < state.compensation)
+        abort("not_enough_compensation")
+    Chain.spend(Contract.creator, amount)
+    ()
```
Comment (thepiwo) [inline test/contracts/channel_whitepaper_example.aes]: no need to return empty tuples

Comment (velzevur) [reply test/contracts/channel_whitepaper_example.aes]: As Hans noted, `unit`s save gas, so this is an optimization

Comment (hanssv) [reply test/contracts/channel_whitepaper_example.aes]: Not here, `Chain.spend` already returns unit :-)

File: test/contracts/channel_whitepaper_example.aes [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    { oracle          = oracle,
+      city            = city,
+      price_per_gen   = price_per_gen,
+      compensation    = compensation,
+      active_from     = 0,
+      active_to       = 0
+      }
+
+  stateful payable entrypoint insure() =
+    if (Call.caller == Contract.creator)
```
Comment (uwiger) [inline test/contracts/channel_whitepaper_example.aes]:
```suggestion
    require (Call.caller != Contract.creator, "service_provider")
```

Comment (uwiger) [reply test/contracts/channel_whitepaper_example.aes]: (would be applied generally to all such tests)

**Pull request merged**